test/instruction/instructions/jump/bconditional.test.ts

```diff
@@ -0,0 +1,448 @@
+    )
+    expect(BMI.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101010011111011)
+    )
+    expect(BPL.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101010111111011)
+    )
+    expect(BVS.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101011011111011)
+    )
+    expect(BVC.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101011111111011)
+    )
+    expect(BHI.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101100011111011)
+    )
+    expect(BLS.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101100111111011)
+    )
+    expect(BGE.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101101011111011)
+    )
+    expect(BLT.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101101111111011)
+    )
+    expect(BGT.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101110011111011)
+    )
+    expect(BLE.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101110111111011)
+    )
+    expect(BAL.encodeInstruction(['label'], offsetMinusTen)[0]).toEqual(
+      Halfword.fromUnsignedInteger(0b1101111011111011)
+    )
+  })
+  test('test encodeInstruction with passing label offset information that is too large', () => {
+    // here only BEQ is used as the other instructions are implemented exactly the same
+    expect(() =>
+      BEQ.encodeInstruction(['label'], negativeOffsetTooBig)
+    ).toThrow()
+    expect(() =>
+      BEQ.encodeInstruction(['label'], positiveOffsetTooBig)
+    ).toThrow()
+  })
+  test('test encodeInstruction with passing label offset information that is not dividable by two', () => {
+    // here only BEQ is used as the other instructions are implemented exactly the same
+    expect(() =>
+      BEQ.encodeInstruction(['label'], negativeOffsetNotDividableByTwo)
+    ).toThrow()
+    expect(() =>
+      BEQ.encodeInstruction(['label'], positiveOffsetNotDividableByTwo)
+    ).toThrow()
+  })
+})
+
+describe('test onExecuteInstruction function for B conditional jump instruction', () => {
+  const firstPCValue = 30
+  const secondPCValue = 20
+
+  test('test execution if flags are correctly set for BEQ', () => {
+    reset()
+    registers.setFlag(Flag.Z, true)
+    execute(BEQ, firstPCValue, secondPCValue)
+  })
+
+  test('test execution if flags are correctly set for BNE', () => {
+    reset()
+    registers.setFlag(Flag.Z, false)
+    execute(BNE, firstPCValue, secondPCValue)
+  })
+
+  test('test execution if flags are correctly set for BCS', () => {
+    reset()
+    registers.setFlag(Flag.C, true)
```

Comment:
Should we check with all this tests if Flags which should not get set are left as is? (Maybe de function changes a wrong flag as well too.)

Reply:
I would argue against this. I mean this tests checks if the condition jumps if the flags are correctly set according to the table in our handout. If you check the instruction code there is never any call to `setFlag()` function on the `Register` class. So in my point of view a test like this is not really worth it.

Reply:
Given that these tests are unit tests and therefor white box tests, it's okay to have the knowledge in the test that there is no `setFlag` and the flags can never be changed.

Reply:
Okay, then we will leave it as it is currently.